subsystem/core/src/main/java/org/wildfly/extension/camel/parser/CamelSubsystemAdd.java

```diff
@@ -57,12 +59,15 @@
 
     public static final int DEPENDENCIES_CAMEL_INTEGRATION = Phase.DEPENDENCIES_LOGGING + 0x01;
     public static final int DEPENDENCIES_CAMEL_WIRINGS = DEPENDENCIES_CAMEL_INTEGRATION + 0x01;
+    public static final int DEPENDENCIES_CAMEL_ENDPOINT_DEPLOYMENT_SCHEDULER = DEPENDENCIES_CAMEL_WIRINGS + 0x01;
 
     public static final int INSTALL_PACKAGE_SCAN_RESOLVER = Phase.INSTALL_WS_DEPLOYMENT_ASPECTS + 0x01;
     public static final int INSTALL_CDI_BEAN_ARCHIVE_PROCESSOR = INSTALL_PACKAGE_SCAN_RESOLVER + 0x01;
     public static final int INSTALL_CAMEL_CONTEXT_CREATE = INSTALL_CDI_BEAN_ARCHIVE_PROCESSOR + 0x01;
     public static final int INSTALL_CONTEXT_ACTIVATION = INSTALL_CAMEL_CONTEXT_CREATE + 0x01;
 
+    public static final int INSTALL_CAMEL_ENDPOINT_DEPLOYER = Phase.INSTALL_WAR_DEPLOYMENT + 0x01;
```

Comment:
What's not clear to me yet, is the consequence of having the endpoint deployer run after the Camel context has been started (I assume that's what happens?).

E.g what happens if I have a route like:

```
from("timer:foo?period=1s&delay=0")
.to("cxfrs://http://foo/bar?resourceClasses=org.foo.bar")
```

Does the Camel context get fully started before the CXF endpoint is deployed. I.e is there some period where the above code would be polling an endpoint that is not yet available?

Maybe it's not an issue (not actually sure whether the current solution deals with this), but it'd be good to check.

Reply:
> Does the Camel context get fully started before the CXF endpoint is deployed?

Yes, I think so, at least for the CDI case. Although I am not sure when exactly the timer:foo?period=1s&delay=0 would start. 

> it'd be good to check.

You mean just investigate what happens when? Or write some sort of a test?

Reply:
It's kind of the same thing. You can investigate by writing a test. We don't necessarily have to commit the test, but I'm interested in what the behaviour is.

Reply:
Here is an excerpt from a log:

```
2018-07-12 09:05:16,462 INFO  [org.apache.camel.cdi.CdiCamelExtension] (MSC service thread 1-3) Camel CDI is starting Camel context [cxfws-secure-cdi-camel-context]
2018-07-12 09:05:16,466 INFO  [org.apache.camel.impl.DefaultCamelContext] (MSC service thread 1-3) Apache Camel 2.22.1-SNAPSHOT (CamelContext: cxfws-secure-cdi-camel-context) is starting
2018-07-12 09:05:16,467 INFO  [org.apache.camel.management.ManagedManagementStrategy] (MSC service thread 1-3) JMX is enabled
2018-07-12 09:05:16,552 INFO  [org.wildfly.extension.camel] (MSC service thread 1-3) Camel context starting: cxfws-secure-cdi-camel-context
2018-07-12 09:05:16,552 INFO  [org.wildfly.extension.camel] (MSC service thread 1-4) Bound camel naming object: java:jboss/camel/context/cxfws-secure-cdi-camel-context
2018-07-12 09:05:17,547 INFO  [org.apache.cxf.wsdl.service.factory.ReflectionServiceFactoryBean] (MSC service thread 1-3) Creating Service {http://subA.secure.ws.cxf.test.camel.wildfly.org/}GreetingServiceService from class org.wildfly.camel.test.cxf.ws.secure.subA.GreetingService
2018-07-12 09:05:18,101 INFO  [org.apache.cxf.endpoint.ServerImpl] (MSC service thread 1-3) Setting the server's publish address to be https://localhost:8443/webservices/greeting-secure-cdi
2018-07-12 09:05:18,335 INFO  [org.apache.camel.impl.DefaultCamelContext] (MSC service thread 1-3) Route: route1 started and consuming from: direct://start
2018-07-12 09:05:18,339 ERROR [org.wildfly.extension.camel] (MSC service thread 1-3) --- scheduling https://localhost:8443/webservices/greeting-secure-cdi: java.lang.RuntimeException
2018-07-12 09:05:18,370 INFO  [org.apache.camel.impl.DefaultCamelContext] (MSC service thread 1-3) Total 6 routes, of which 6 are started
2018-07-12 09:05:18,371 INFO  [org.apache.camel.impl.DefaultCamelContext] (MSC service thread 1-3) Apache Camel 2.22.1-SNAPSHOT (CamelContext: cxfws-secure-cdi-camel-context) started in 1.906 seconds
2018-07-12 09:05:18,415 ERROR [org.wildfly.extension.camel] (MSC service thread 1-6) --- deploying https://localhost:8443/webservices/greeting-secure-cdi: java.lang.RuntimeException
2018-07-12 09:05:18,467 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-6) WFLYUT0021: Registered web context: '/webservices/greeting-secure-cdi' for server 'default-server'
2018-07-12 09:05:18,468 INFO  [org.wildfly.extension.camel] (MSC service thread 1-6) Add Camel endpoint: http://127.0.0.1:8080/webservices/greeting-secure-cdi
```

Note that the endpoint is exposed at `09:05:18,467` from `MSC service thread 1-6` 96 milliseconds after the camel context started at `09:05:18,371` inside `MSC service thread 1-3`. This is quite a fast machine.

Do you think we should address this somehow?

Reply:
It is a change in behaviour compared to how things work now.

My example is a bit of an edge case in some respects. However, if the Camel context state == Started, then I'd expect all of the relevant endpoints to be fully up and running at that point.

It's a tricky problem. Not sure if we want to put significant effort into resolving it now. We could create an issue and address it later.

@tdiesler what do you think?

Reply:
+1 create an issue and address it later

Reply:
I'd say, if it is not obviously wrong we can get it in and if there should be a problem with it in the future, we deal with it then.

Reply:
Filed https://github.com/wildfly-extras/wildfly-camel/issues/2600